syft/pkg/cataloger/java/archive_parser.go

```diff
@@ -171,34 +171,36 @@ func (j *archiveParser) discoverMainPackage() (*pkg.Package, error) {
 		return nil, nil
 	}
 
-	archiveCloser, err := os.Open(j.archivePath)
-	if err != nil {
-		return nil, fmt.Errorf("unable to open archive path (%s): %w", j.archivePath, err)
-	}
-	defer archiveCloser.Close()
-
 	// grab and assign digest for the entire archive
-	digests, err := intFile.NewDigestsFromFile(archiveCloser, javaArchiveHashes)
+	digests, err := getDigestsFromArchive(j.archivePath)
 	if err != nil {
-		log.Warnf("failed to create digest for file=%q: %+v", j.archivePath, err)
+		return nil, err
 	}
 
+	licenses := make([]pkg.License, 0)
 	// we use j.location because we want to associate the license declaration with where we discovered the contents in the manifest
-	licenses := pkg.NewLicensesFromLocation(j.location, selectLicenses(manifest)...)
+	locationLicenses := pkg.NewLicensesFromLocation(j.location, selectLicenses(manifest)...)
+	if locationLicenses != nil {
+		licenses = append(licenses, locationLicenses...)
+	}
 	/*
 		We should name and version from, in this order:
 		1. pom.properties if we find exactly 1
 		2. pom.xml if we find exactly 1
 		3. manifest
 		4. filename
 	*/
-	name, version := j.guessMainPackageNameAndVersionFromPomInfo()
+	name, version, pomLicenses := j.guessMainPackageNameAndVersionFromPomInfo()
 	if name == "" {
 		name = selectName(manifest, j.fileInfo)
 	}
 	if version == "" {
 		version = selectVersion(manifest, j.fileInfo)
 	}
+	if len(pomLicenses) > 0 {
```

Comment:
Thanks a lot @spiffcs ! I just have a query about this line, in my PR the pomLicenses were only added if we didn't already get licenses from the Bundle-License. I'm going to contribute PRs after this is merged to reach out to Maven Central to retrieve the license if not contained in the pom, which we probably want to avoid if we already have a license from the Manifest.

Reply:
Oh interesting! cc @wagoodman for the morning - we discussed this yesterday. I thought that it might be best to surface all licenses found regardless of source to provide the most information to the user (manifest or pom). @coheigea's comment makes sense though with respect to the other PRs coming. Leaving this open for @wagoodman comments

Reply:
is there a reason to make this mutually exclusive? Typically we raise up as much information as is useful, and I feel that license info should be fairly unbiased. That is, if there are two licenses then knowing that they agree or disagree is useful.

Reply:
We've been talking about the Location enhancement for a while but I couldn't find an issue for it -- created https://github.com/anchore/syft/issues/2211 to track (and linking with this comment 😄 )